controllers/horizon_controller.go

```diff
@@ -340,6 +344,23 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 		return ctrl.Result{}, err
 	}
 
+	if instance.Spec.SharedMemcached == "" {
+		memcachedExists, err := r.getMemcachedCR(ctx, fmt.Sprintf("%s-memcached", instance.Name))
+		if err != nil {
+			r.Log.Info("Failed retrieving Memcached CR list")
+			return ctrl.Result{}, err
+		}
+		if !memcachedExists {
+			memcached := r.renderMemcached(instance)
+			err = r.Client.Create(ctx, memcached)
```

Comment:
I suggest to use the pattern of CreateOrPatch instead of trying to Get and if not exists Create the CR. CreateOrPatch will help also in case only some part of the Memcached instance needs to be updated (e.g. when instance.Spec.Replicas, changes). See for example https://github.com/openstack-k8s-operators/lib-common/blob/e77d8d19c2ffab69535fbbd47531b833d82fee49/modules/common/deployment/deployment.go#L57

Reply:
Done in the latest update.